docs/keps/20200619-federated-resource-status.md

```diff
@@ -0,0 +1,537 @@
+---
+kep-number: 20200619
+short-desc: Kubefed -- Federated Resources Status
+title: Kubefed -- Federated Resources Status
+authors:
+  - "@hectorj2f"
+reviewers:
+  - "@irfan"
+  - "@hectorj2f"
+  - "@jimmidyson"
+  - "@pmorie"
+approvers:
+- "@irfan"
+- "@jimmidyson"
+- "@pmorie"
+editor: TBD
+creation-date: 2020-06-19
+last-updated: 2020-07-28
+status: provisional
+---
+
+# Kubefed -- Federated Resources Status
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+Kubefed needs to improve its definition of status for the federated resources.
+Users lack of proper visibility over the status of the federated
+resources. For instance, if you federated a deployment the federated status should
+report if the deployment failed or error at any time.
+
+A federated resource only reflects the status of propagation actions, but it doesn't
+reflect the status if whether the resource is running or failed.
+
+## Motivation
+
+Nowadays users have to connect to the kubefed clusters to be aware if the federated
+resources are healthy or not across clusters.
+
+### Goals
+
+* Quickly identify unhealthy federated resources by relying on the status of the federated resources.
+* Improve the troubleshooting of failures when propagating resources.
+* Define a new api version potentially `v1beta2` to include these new fields.
```

Comment:
Which API will be promoting to `v1beta2`?

Reply:
The new properties added to the FederatedTypeConfig should bump the api version.